### PR TITLE
[App Interface Reporter] Collect info for multi env merge activities 

### PR DESCRIPTION
**Pull request by violetyao**
Jira: https://issues.redhat.com/browse/APPSRE-2954
Issue: Currently merges_to_master only collects activities in master branch, while it is also possible to have activities in main, tag, etc. 
Example (`merge_activities` is not null):
```
$schema: /app-sre/report-1.yml
labels:
  app: quayio
name: quayio-2021-03-12
app:
  $ref: /services/quayio/app.yml
date: '2021-03-12'
contentFormatVersion: 1.0.0
content: |
  promotions:
    quay-production-standby:
      total: 1
      success: 1
  merge_activities:
    https://github.com/quay/quay:
    - branch: master
      total: 1
      success: 0
    https://github.com/app-sre/syslog-cloudwatch-bridge:
    - branch: master
      total: 1
      success: 1
    https://github.com/quay/clair:
    - branch: master
      total: 10
      success: 10
    https://github.com/coreos-inc/apostille:
    - branch: master
      total: 1
      success: 0
    https://github.com/quay/update-banner-job:
    - branch: master
      total: 10
      success: 7
  container_vulnerabilities: null
  post_deploy_jobs:
  - cluster: quays02ue1
    namespace: apostille
    post_deploy_job: false
  - cluster: quayp05ue1
    namespace: quay
    post_deploy_job: false
  - cluster: quayp03ue1
    namespace: quay-mysql-diag-prod
    post_deploy_job: false
  - cluster: quayp04ue2
    namespace: quay
    post_deploy_job: false
  deployment_validations: null
```

Example (`merge_activities` is null):
```
$schema: /app-sre/report-1.yml
labels:
  app: che-dev
name: che-dev-2021-03-12
app:
  $ref: /services/che-dev/app.yml
date: '2021-03-12'
contentFormatVersion: 1.0.0
content: |
  promotions: null
  merge_activities: null
  container_vulnerabilities: null
  post_deploy_jobs: null
  deployment_validations: null

```

### CODE REVIEW

File: tools/app_interface_reporter.py [hunk 1]
```diff
@@ -387,10 +403,30 @@ def get_build_history(jenkins_map, jobs, timestamp_limit):
...
             logging.info(f"getting build history for {job['name']}")
-            build_history = \
-                jenkins.get_build_history(job['name'], timestamp_limit)
-            repo_url = get_repo_url(job)
-            history[repo_url] = build_history
+            # continue if no repo_url
+            if 'properties' not in job:
+                continue
+            if 'github' not in job['properties'][0]:
+                continue
```
Comment (BumbleFeng) [inline tools/app_interface_reporter.py:410]:
```
try:
  repo_url = get_repo_url(job)
except KeyError:
  continue
```

File: tools/app_interface_reporter.py [hunk 2]
```diff
@@ -387,10 +403,30 @@ def get_build_history(jenkins_map, jobs, timestamp_limit):
...
+
+            if 'build_branch' in job:
+                job_env = job['build_branch']
+            elif 'id' in job:
+                job_env = job['id']
+            else:
+                continue
```
Comment (BumbleFeng) [inline tools/app_interface_reporter.py:417]: Could you give some examples of these 3 kinds of jobs?

File: tools/app_interface_reporter.py (hunk 1 above)
Comment (violetyao) [reply tools/app_interface_reporter.py:410]: Got it!

File: tools/app_interface_reporter.py (hunk 2 above)
Comment (violetyao) [reply tools/app_interface_reporter.py:417]: `ci-int` and `ci-ext` jobs use `id` 
Example
```
'name': 'insights-platform-pes-cloud-gl-build-master',
'properties': [{'github': {'url': 'https://gitlab.cee.redhat.com/insights-platform/pes-cloud.git'}},
  {'gitlab': {'connection': 'gitlab cee'}}],
'id': 'gl-build-master'
```

while `ci-centos` use `build_branch`
Example:
```
'name':  'devtools-fabric8-notification-build-master-coverage',
'properties': [{'github': {'url': 'https://github.com/fabric8-services/fabric8-notification/'}}],
'build_branch': 'master'
```

Currently we do not have the case where neither `id` nor `build_branch` appears.

Comment (BumbleFeng) [reply tools/app_interface_reporter.py:417]: Could you try with job['branch']?

Comment (violetyao) [reply tools/app_interface_reporter.py:417]: Perfect! All jobs has `branch` key but it seems that there are some small variations. For example, 
```
'instance' : ci-int
'name': dtsd-osio-scan-gl-build-master
'branch':  master
'id': gl-build-master-osio-scan
```
and 
```
'instance' : ci-ext
'name': app-sre-manifest-bouncer-gh-build-tag
'branch': refs/tags/*
'id': gh-build-tag
```
Do we care about those variations or shall we just go ahead and use job['branch']?

Comment (BumbleFeng) [reply tools/app_interface_reporter.py:417]: id is used for matching with job-temple. It is not unique and it may not reflect the actual branch.

Comment (violetyao) [reply tools/app_interface_reporter.py:411]: updated!

**Review by BumbleFeng — APPROVED** [review]

**Pull request merged**